hazelcast/proxy/vector_collection.py

```diff
@@ -99,10 +224,41 @@ def search_near_vector(
         )
 
     def remove(self, key: Any) -> Future[Optional[Document]]:
+        """Removes the mapping for a key from this VectorCollection if it is present.
+
+        The VectorCollection will not contain a mapping for the specified key once the call
+        returns.
+
+        Warning:
+            This method uses ``__hash__`` and ``__eq__`` methods of binary form
+            of the key, not the actual implementations of ``__hash__`` and
+            ``__eq__`` defined in key's class.
+
+        Args:
+            key: Key of the mapping to be deleted.
+
+        Returns:
+            The Document associated with key, or ``None`` if there was
+            no mapping for key.
+        """
         check_not_none(key, "key can't be None")
         return self._remove_internal(key)
 
     def delete(self, key: Any) -> Future[None]:
+        """Removes the mapping for a key from this VectorCollection if it is present
+        (optional operation).
```

Comment:
what does it mean "optional operation" in this context? It is used also for in other proxies

Reply:
I took it from the `Map` documentation. I think it means the operation is no-op if the mapping is not present.

Reply:
`delete` deletes a mapping, it's not related to the proxy lifetime.
Does `VectorCollection.delete` behave differently than `Map.delete` if the mapping doesn't exist?

Reply:
> Does VectorCollection.delete behave differently than Map.delete if the mapping doesn't exist?

no, it is the same . I think I misunderstood before: you meant mapping=entry, I was thinking mapping=IMap.

I am not sure how to concisely say that the operation does nothing if there is nothing to do, but this is not what I understand when I read "optional operation". I think "idempotent" would be more precise.

and why `delete` is "optional" while `remove` is not? because the latter returns a value? (BTW, `remove` is not idempotent, because repeated it will return different result).

Reply:
As far as I see from the server-side code, the difference between `remove` and `delete` may be related to map stores. `remove` calls `mapDataStore.load(key)` even if a mapping is not found.

Reply:
Btw, I don't think `delete` is idempotent either.
Apart from that, even calls like `get` are not truly idempotent in Hz.

Reply:
vector collection does not support map stores, so there is no difference between `remove` and `delete` that I would call "optional operation". So I'd remove `optional operation` from vector docs not to introduce confusion.

Reply:
I've added `optional operation` to the `vc.remove` method for consistency.

Reply:
for consistency with what? Vector colleciton is not IMap and "optional operation" does not mean anything in context of vector collection. We should remove it because it is only misleading and confusing in case of vector collection.

Reply:
`VectorCollection` is closest to the `IMap. As we've discussed, "optional operation" means the operation becomes no-op if the mapping does not exist. The user wouldn't necessarily know whether VectorCollection uses a map store or not.

Feel free to send another PR to explain this better after this PR is merged.

Reply:
> "optional operation" means the operation becomes no-op if the mapping does not exist

this is completely not obvious meaning for me, do you think that our users will understand it that way? @shultseva @vbekiaris how do understand that?

also:

1. there are more operations that do not change state of the system if the collection does not exist, eg. `get`, `search`, `size`, `clear`
2. for all those cases it does not matter if the collection exists or not (strictly speaking if `DistributedObject` exists) but if the collection (or partition for `get`) is empty or not (collection that does not exist is empty by definition). And the result is logically obvious: if the collection is empty then removing from it does not change it
3. we want to cut ties between vector collection and IMap, there is no value is artificially making them similar

so this is a useless, confusing information.